pkg/strategy/xdepthmaker/strategy.go

```diff
@@ -567,17 +571,53 @@ func (s *Strategy) CrossRun(
 		ID, s.InstanceID()); err != nil {
 		return err
 	}
-
+	// initialize source book
 	sourceMarketStream := s.hedgeSession.Exchange.NewStream()
 	sourceMarketStream.SetPublicOnly()
 	sourceMarketStream.Subscribe(types.BookChannel, s.HedgeSymbol, types.SubscribeOptions{
 		Depth: types.DepthLevelFull,
 		Speed: types.SpeedLow,
 	})
-
 	s.sourceBook = types.NewStreamBook(s.HedgeSymbol, s.hedgeSession.ExchangeName)
 	s.sourceBook.BindStream(sourceMarketStream)
 
+	// initialize maker book
+	makerMarketStream := s.makerSession.Exchange.NewStream()
+	makerMarketStream.SetPublicOnly()
+	makerMarketStream.Subscribe(types.BookChannel, s.Symbol, types.SubscribeOptions{
+		Depth: types.DepthLevelFull,
+		Speed: types.SpeedLow,
+	})
+	s.makerBook = types.NewStreamBook(s.Symbol, s.makerSession.ExchangeName)
+	s.makerBook.BindStream(makerMarketStream)
+	s.makerBook.OnUpdate(func(_ types.SliceOrderBook) {
```

Comment:
you can pull this closure out to a method of "Strategy"

something like: `s.makerBook.OnUpdate(s.handleMakerBookUpdate)`

Reply:
Done.